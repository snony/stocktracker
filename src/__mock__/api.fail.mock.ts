export const getNews = (_: string) => {
    return Promise.reject(new Error('fail'))
}

export default { getNews }
export const mockApiFailFetch = {
    getCompanySymbols: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
    getHistory: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
    getNews: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
    getKeyStats: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
    getOverview: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
    getPeers: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data'))
}