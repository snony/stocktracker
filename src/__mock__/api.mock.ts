import { mockGlobalState } from './globalstate.mock'

export const mockApi = {
  getCompanySymbols: jest.fn().mockResolvedValue(mockGlobalState.companySymbolsState),
  getHistory: jest.fn().mockResolvedValue(mockGlobalState.history),
  getNews: jest.fn().mockResolvedValue(mockGlobalState.news),
  getKeyStats: jest.fn().mockResolvedValue(mockGlobalState.keystats),
  getOverview: jest.fn().mockResolvedValue(mockGlobalState.overview),
  getPeers: jest.fn().mockResolvedValue(mockGlobalState.peers)
}

export const mockFailedApi = {
  getCompanySymbols: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
  getHistory: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
  getNews: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
  getKeyStats: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
  getOverview: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data')),
  getPeers: jest.fn().mockRejectedValueOnce(new Error('Cannot fetch data'))
}
