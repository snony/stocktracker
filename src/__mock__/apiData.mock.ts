export const mockFetchData = { data: { company: 'Apple Inc.', symbol: 'AAPL' } }

export const keyStatsInput = {
  quote: {
    previousClose: 3,
    open: 4,
    high: 5,
    low: 6,
    latestVolume: 7,
    avgTotalVolume: 8,
    peRatio: 9,
    marketCap: 10,
    week52High: 11,
    week52Low: 12
  },
  stats: {
    latestEPS: 1,
    dividendYield: 2
  }
}

export const historyInput = [
  {
    open: 135,
    date: '14/09/2018'
  },
  {
    open: 140,
    date: '15/09/2018'
  },
  {
    open: 145,
    date: '16/09/2018'
  }
]

export const historyInput1D = [
  {
    open: 135,
    date: '14/09/2018',
    minute: '13:00'
  },
  {
    open: 136,
    date: '14/09/2018',
    minute: '13:05'
  },
  {
    open: 137,
    date: '14/09/2018',
    minute: '13:10'
  }
]
