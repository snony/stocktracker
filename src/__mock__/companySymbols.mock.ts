import { Company } from '../types'

const companySymbols: Company[] = [
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Facebook Inc', symbol: 'fb' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' },
  { name: 'Apple Inc', symbol: 'aapl' }
]

export const getCompanySymbols = (amount: number = 5) => companySymbols.slice(0, amount)
