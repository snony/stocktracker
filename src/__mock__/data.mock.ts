import { KeyStats } from "keystats/types";

export const mockFetchData = { data: { company: 'Apple Inc.', symbol: 'AAPL' } }

export const keyStatsInput = {
  quote: {
    previousClose: 3,
    open: 4,
    high: 5,
    low: 6,
    latestVolume: 7,
    avgTotalVolume: 8,
    peRatio: 9,
    marketCap: 10,
    week52High: 11,
    week52Low: 12
  },
  stats: {
    latestEPS: 1,
    dividendYield: 2
  }
}

export const historyInput = [
  {
    open: 135,
    date: '14/09/2018'
  },
  {
    open: 140,
    date: '15/09/2018'
  },
  {
    open: 145,
    date: '16/09/2018'
  }
]

export const historyInput1D = [
  {
    open: 135,
    date: '14/09/2018',
    minute: '13:00'
  },
  {
    open: 136,
    date: '14/09/2018',
    minute: '13:05'
  },
  {
    open: 137,
    date: '14/09/2018',
    minute: '13:10'
  }
]

export const mockKeyStats: KeyStats = {
  previousClose: 10,
  dayHigh: 100,
  dayLow: 10,
  volume: 10000,
  marketCap: 100000000,
  peRatio: 10,
  open: 10,
  week52High: 100,
  week52Low: 10,
  avgTotalVolume: 10000,
  earningsPerShare: 10,
  dividendYield: 10
}

export const mockPeers = ['peers1', 'peers2', 'peers3', 'peers4']

export const mockGlobalState = {
  companySymbols: [] as any,
  company: {},
  history: {},
  news: [] as any,
  keystats: mockKeyStats,
  overview: {},
  peers: mockPeers
}

