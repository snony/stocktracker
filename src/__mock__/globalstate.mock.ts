import { GlobalState } from 'types'

import { companySymbolsMock, mockCompanySymbols } from './companySymbols.mock'
import { mockFailedHistoryState, mockHistoryState } from './history.mock'
import { mockFailedKeyStatsState, mockKeyStatsState } from './keystats.mock'
import { mockNews } from './news.mock'
import { mockFailedOverviewData, mockOverviewData } from './overview.mock'
import { mockFailedPeersState, mockPeersState } from './peers.mock'

export const mockGlobalState: GlobalState = {
  companySymbolsState: mockCompanySymbols,
  company: companySymbolsMock[0],
  history: mockHistoryState,
  news: mockNews,
  keystats: mockKeyStatsState,
  overview: mockOverviewData,
  peers: mockPeersState
}

export const mockFailedGlobalState = {
  history: mockFailedHistoryState,
  keystats: mockFailedKeyStatsState,
  overview: mockFailedOverviewData,
  peers: mockFailedPeersState
} as GlobalState
