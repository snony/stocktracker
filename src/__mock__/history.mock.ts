import { filters } from "history/historyConst";
import { initialState } from "history/historyReducer";
import { HistoryState } from "history/types";

export const mockChartData = [
  { date: '12/09/2018', price: 13.5 },
  { date: '13/09/2018', price: 14.5 },
  { date: '14/09/2018', price: 12.5 }
]

export const mockHistory = (overrides: HistoryState): HistoryState => ({
  ...initialState,
  ...overrides
})

export const mockHistoryState = mockHistory({
  history: mockChartData,
  dateFilter: filters.YTD,
  priceFilter: filters.CLOSE
})
