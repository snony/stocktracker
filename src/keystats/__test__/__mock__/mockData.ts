export const mockKeyStats = {
  previousClose: 10,
  dayHigh: 100,
  dayLow: 10,
  volume: 10000,
  marketCap: 100000000,
  peRatio: 10,
  open: 10,
  week52High: 100,
  week52Low: 10,
  avgTotalVolume: 10000,
  earningsPerShare: 10,
  dividendYield: 10
}
