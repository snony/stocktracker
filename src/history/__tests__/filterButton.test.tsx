import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import FilterButton from 'history/filterButton'
import React from 'react'
import renderer from 'react-test-renderer'

import { mockFilterButtonProps } from './__mock__/mockData'

Enzyme.configure({ adapter: new Adapter() })

describe('filterButton component', () => {
  it('should render unselected filter button correctly', () => {
    const tree = renderer.create(<FilterButton {...mockFilterButtonProps} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render selected filter button correctly', () => {
    const tree = renderer.create(<FilterButton {...mockFilterButtonProps} selected />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should call onClick method when button is clicked', () => {
    const wrapper = shallow(<FilterButton {...mockFilterButtonProps} />)

    wrapper.find('button').simulate('click')
    expect(mockFilterButtonProps.onClick).toHaveBeenCalledTimes(1)
  })
})
