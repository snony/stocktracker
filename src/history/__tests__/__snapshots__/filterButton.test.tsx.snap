// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterButton component should render selected filter button correctly 1`] = `
<button
  className="filter-button filter-button--selected"
  onClick={[Function]}
>
  OPEN
</button>
`;

exports[`filterButton component should render unselected filter button correctly 1`] = `
<button
  className="filter-button "
  onClick={[Function]}
>
  OPEN
</button>
`;
